backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/lowlevel/index/mapping/impl/PropertyMapping.java

```diff
@@ -86,6 +86,52 @@ public class PropertyMapping extends AbstractTypeMapping {
 	@SerializedName("term_vector")
 	private String termVector;
 
+	/*
+	 * Dense vector datatype
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html
+	 */
+	@SerializedName("element_type")
+	private String elementType;
+
+	/*
+	 * Dense vector datatype
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html
+	 */
+	private Integer dims;
+
+	/*
+	 * Dense vector datatype
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html
+	 */
+	private String similarity;
+
+	/*
+	 * Dense vector datatype
+	 * https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html
+	 */
+	@SerializedName("index_options")
+	private JsonElement indexOptions;
+
+	/*
+	 * k-NN vector datatype
+	 * https://opensearch.org/docs/latest/field-types/supported-field-types/knn-vector/
+	 */
+	private Integer dimension;
+
+	/*
+	 * k-NN vector datatype
+	 * https://opensearch.org/docs/latest/field-types/supported-field-types/knn-vector/
+	 */
+	private JsonElement method;
+
+	/*
+	 * k-NN vector datatype
+	 * https://opensearch.org/docs/latest/field-types/supported-field-types/knn-vector/
+	 */
+	@SerializedName("data_type")
+	private String dataType;
+
+
```

Comment:
If you add stuff here, I think you need to also add validation rules to `PropertyMappingValidator`. And maybe some tests :x

Reply:
Might be worth adding another warning about this at the top of this file :/